src/btree/bt_read.c

```diff
@@ -68,6 +68,129 @@ __row_instantiate(WT_SESSION_IMPL *session,
 	return (0);
 }
 
+/*
+ * __wt_find_lookaside_upd --
+ * 	Scan the lookaside for a record the btree cursor wants to position on.
+ * 	Create an update for the record and return to the caller.
+ */
+int
+__wt_find_lookaside_upd(WT_SESSION_IMPL *session,
+    WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool ignore_birthmark)
+{
+	WT_CACHE *cache;
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_ITEM *key;
+	WT_ITEM las_key, las_value;
+	WT_REF *ref;
+	WT_UPDATE *upd;
+	wt_timestamp_t durable_timestamp, las_timestamp;
+	size_t incr;
+	uint64_t las_counter, las_pageid, las_txnid;
+	uint32_t las_id, session_flags;
+	uint8_t prepare_state, upd_type;
+	int cmp;
+	bool locked, upd_visible;
+
+	cache = S2C(session)->cache;
+	cursor = NULL;
+	session_flags = 0;		/* [-Werror=maybe-uninitialized] */
+	ref = cbt->ref;
+	las_pageid = ref->page_las->las_pageid;
+	key = &cbt->iface.key;
+	*updp = upd = NULL;
+
+	/* Open a lookaside table cursor. */
+	__wt_las_cursor(session, &cursor, &session_flags);
+
+	/*
+	 * The lookaside records are in key and update order, that is, there
+	 * will be a set of in-order updates for a key, then another set of
+	 * in-order updates for a subsequent key. We scan through all of the
+	 * updates till we locate the one we want.
+	 */
+	WT_PUBLISH(cache->las_reader, true);
+	__wt_readlock(session, &cache->las_sweepwalk_lock);
+	WT_PUBLISH(cache->las_reader, false);
+	locked = true;
```

Comment:
This looks wrong to me: `WT_PUBLISH` ensures that all previous writes have completed before the current write, and it's unclear to me what previous writes are being flushed.

Reply:
I was not sure we need to publish here myself. I based this code on ```__las_page_instantiate()```, are we publishing there for a reason (maybe related to generation number updates in ```__las_page_instantiate_verbose()```) ?

Can you please have a look, I will remove the publish then, it does seem they aren't needed.

Reply:
@sueloverso, in https://github.com/wiredtiger/wiredtiger/commit/f29d2b9a20662d60baadda142c2283d7afc18b33 you and @michaelcahill wrapped a lock in two `WT_PUBLISH` statements in `__las_page_instantiate()`:
```
WT_PUBLISH(cache->las_reader, true);
__wt_readlock(session, &cache->las_sweepwalk_lock);
WT_PUBLISH(cache->las_reader, false);
```
I don't see why either of the `WT_PUBLISH` statements would be necessary, and I can't find any discussion. Can you reconstruct what's going on here?

(Sorry, I know this change is old, but @sulabhM copied in his new code, and I noticed it there.)

Reply:
It has been a long time. My best guess is that the intent is that the value of `cache->las_reader` needs to be immediately visible to other threads. That field will cause a long LAS sweep to break out of its sweep-work if it knows a reader is waiting.

Reply:
OK, here's my argument:

The first `WT_PUBLISH` would flush any writes before setting `las_reader`. I can't identify any writes before the set of `las_reader` that need to be flushed.

The second `WT_PUBLISH` would flush any writes after the `wt_readlock()` call before setting `las_reader`. There aren't any, so again, it's not doing anything useful.

@sulabhM, unless you or @sueloverso see something I'm missing, I think we should change both instances to:
```
cache->las_reader = true;
__wt_readlock(session, &cache->las_sweepwalk_lock);
cache->las_reader = false;
```

Reply:
@keithbostic and I discussed and agreed to this.